internal/hcs/service.go

```diff
@@ -0,0 +1,39 @@
+//go:build windows
+
+package hcs
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/Microsoft/hcsshim/internal/interop"
+	hcsschema "github.com/Microsoft/hcsshim/internal/schema2"
+)
+
+// GetServiceProperties returns properties of the host compute service.
+func GetServiceProperties(ctx context.Context, q hcsschema.PropertyQuery) (properties *hcsschema.ServiceProperties, err error) {
+	operation := "hcs::GetServiceProperties"
+
+	queryb, err := json.Marshal(q)
+	if err != nil {
+		return nil, err
+	}
+
+	var resultp, propertiesp *uint16
+	err = hcsGetServicePropertiesContext(ctx, string(queryb), &propertiesp, &resultp)

```

Comment:
>hcsGetServicePropertiesContext [](http://example.com/codeflow?start=7&length=30)

HcsGetServiceProperties is a sync function, no need to invoke it asynchronously.

Reply:
fix